### PR TITLE
Use a separate endpoint for AI settings

### PR DESCRIPTION
**Pull request by marcospri**
This matches the UI and makes for a smaller, more focused view.

## Testing

- General values 

Go to http://localhost:8001/admin/instance/8/

Update any values over the info tab, click save and check the values are indeed updated.

- Settings
Again in  http://localhost:8001/admin/instance/8/

Update any settings and click save at the bottom. The settings are updated.

### CODE REVIEW

File: lms/templates/admin/application_instance/show.html.jinja2 [hunk 1]
```diff
@@ -108,7 +108,7 @@
                 </li>
                 <li>
                     <form method="POST"
-                          action="{{ request.route_url("admin.instance", id_=instance.id) }}">
+                          action="{{ request.route_url("admin.instance.settings", id_=instance.id) }}">
```
Comment (marcospri) [inline lms/templates/admin/application_instance/show.html.jinja2:111]: Submit this form to the new endpoint

File: lms/views/admin/application_instance/update.py [hunk 2]
```diff
@@ -51,7 +49,34 @@ def update_instance(self):
             developer_secret=self.request.params.get("developer_secret", "").strip(),
         )
 
+        # Notes are displayed in the main info tab but are stored alongside notes
+        notes = self.request.params.get("hypothesis.notes")
```
Comment (marcospri) [inline lms/views/admin/application_instance/update.py:53]: This is a bit of wart here but I it makes a lot of sense UI wise.

File: lms/views/admin/application_instance/update.py [hunk 3]
```diff
@@ -51,7 +49,34 @@ def update_instance(self):
...
+
+        self.application_instance_service.update_application_instance(
+            ai,
+            developer_key=self.request.params.get("developer_key", "").strip(),
+            developer_secret=self.request.params.get("developer_secret", "").strip(),
+        )
+
         for field in ApplicationSettings.fields:
+            # Notes are updated in the main `info` tab, skip it here
+            if field.compound_key == "hypothesis.notes":
```
Comment (marcospri) [inline lms/views/admin/application_instance/update.py:77]: The other side of the wart.

File: tests/unit/lms/views/admin/application_instance/update_test.py [hunk 4]
```diff
@@ -105,11 +119,10 @@ def test_update_instance_saves_ai_settings(
     ):
         pyramid_request.params[f"{setting}.{sub_setting}"] = value
 
-        views.update_instance()
+        views.update_instance_settings()
```
Comment (marcospri) [inline tests/unit/lms/views/admin/application_instance/update_test.py:122]: Tests were almost already separated between settings and the rest of the fields.

**Review by acelaya — APPROVED** [review]

**Pull request merged**